admin/section/class-convertkit-settings-base.php

```diff
@@ -144,11 +144,15 @@ public function get_forms() {
 		$forms = get_transient( 'convertkit_forms' );
 
 		if ( false === $forms ) {
-			$this->api->update_resources( $this->options['api_key'], $this->options['api_secret'] );
 
-			$forms = get_option( 'convertkit_forms' );
+			if ( ! empty( $this->options['api_key'] ) && ! empty( $this->options['api_secret'] ) ) {
```

Comment:
Are `api_key` and `api_secret` always both set?

Reply:
@bradpauly In theory, yes, but there have been a number of undefined-index notices for these that have happened on customer sites. And in any event that they're _not_ set, the rest of the code shouldn't run anyways, so suppressing those notices won't interfere with functionality at all.